keps/sig-storage/2485-read-write-once-pod-pv-access-mode/README.md

```diff
@@ -0,0 +1,972 @@
+capability bits for this access mode.
+
+```protobuf
+      // Indicates the SP supports the SINGLE_NODE_MULTI_WRITER access
+      // mode.
+      SINGLE_NODE_MULTI_WRITER = 13;
+```
+
+Put more succinctly:
+
+|                  | Driver Supports `SINGLE_NODE_*_WRITER` | Driver Does Not Support `SINGLE_NODE_*_WRITER` |
+|------------------|----------------------------------------|------------------------------------------------|
+| ReadWriteOncePod | SINGLE_NODE_SINGLE_WRITER              | Depends on the CSI driver                      |
+| ReadWriteOnce    | SINGLE_NODE_MULTI_WRITER               | SINGLE_NODE_WRITER (Existing behavior)         |
+
+Example CSI clients that will need changing are kubelet, the
+external-provisioner and external-attacher.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+See the [version skew strategy] section below for additional scenarios.
+
+[version skew strategy]: #version-skew-strategy
+
+#### ReadWriteOncePod PVC Used Twice Fails for Second Consumer
+
+This scenario asserts a ReadWriteOncePod can only be bind mounted into a single
+pod on a single node.
+
+- User creates a PVC with ReadWriteOncePod access mode
+- User creates pod 1 using this PVC, scheduled on node 1
+- User creates pod 2 using this PVC, also scheduled on node 1
+- User observes pod 2 fails to start because the referenced PVC is in-use by
+  another pod on the same node
+
+Additionally, for attachment:
+
+- User creates pod 3 using this PVC, scheduled on node 2
+- User observes pod 3 fails to start because the referenced PVC is attached to
+  another node
+
+#### ReadWriteOnce PVC Continues to Succeed with New Kubernetes, Old CSI Driver
+
+This scenario asserts the existing ReadWriteOnce behavior is preserved for old
+CSI drivers. The exact behavior may differ across CSI drivers since not all
+drivers conform to the CSI spec, but it should be consistent with how it behaved
+before.
+
+- User creates a PVC with ReadWriteOncePod access mode
+- User creates pod 1 using this PVC, scheduled on node 1
+- User observes pod running
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
```

Comment:
Does PV controller need modification?

Reply:
I don't see any required changes. The only access mode logic is checking if the PV's access modes are a superset of the PVC's access modes.